cmd/md5sum/md5sum.go

```diff
@@ -27,9 +28,22 @@ By default, the hash is requested from the remote.  If MD5 is
 not supported by the remote, no hash will be returned.  With the
 download flag, the file will be downloaded from the remote and
 hashed locally enabling MD5 for any remote.
+
+This command can also hash data received on standard input (stdin),
+by not passing a remote:path, or by passing a hyphen as remote:path
+when there is data to read (if not, the hypen will be treated literaly,
+as a relative path).
 `,
-	Run: func(command *cobra.Command, args []string) {
-		cmd.CheckArgs(1, 1, command, args)
+	RunE: func(command *cobra.Command, args []string) error {
+		cmd.CheckArgs(0, 1, command, args)
+		if len(args) == 0 {
```

Comment:
You could factor these 8 lines into hashsum as they are repeated 3 times so it is probably worth it. (Pass in `startArg` which will be 0 here and 1 in hashsum).

Can we give it a good enough name is the questions?